examples/resnet_cifar10.py

```diff
@@ -0,0 +1,234 @@
+'''
+Pai Peng (pengpai_sh@163.com)
+
+An re-implementation of Residual Network (ResNet) on CIFAR-10 dataset in paper "Deep Residual Learning for
+Image Recognition". The Reisdual Blocks refers to
+ https://github.com/keunwoochoi/residual_block_keras/blob/master/residual_blocks.py
+(Keunwoo Choi).
+Note that his implementation has already used "pre-activation" proposed in the new version of ResNet
+paper "Identity Mappings in Deep Residual Networks".
+
+1. Set n = 3 to obtain a 20-layer ResNet, testing accuracy is 89.4% (loss: 0.31668),
+   training with 36 hours (75 epochs).
+2. Set n = 5 to obtain a 32-layer ResNet, testing accuracy is 90.8% (loss: 0.28545),
+   training with 58 hours (75 epochs).
+3. Set n = 8 to obtain a 50-layer ResNet, testing accuracy is 90.89% (loss: 0.28211),
+   training with 94 hours (75 epochs).
+
+'''
+
+import sys
+sys.setrecursionlimit(99999)
+import numpy as np
+np.random.seed(2016)  # for reproducibility
+import keras
+import keras.backend as K
+from keras.models import Sequential
+from keras.layers.core import Dense, Dropout, Activation, Flatten
+from keras.layers.convolutional import Convolution2D, MaxPooling2D, ZeroPadding2D, AveragePooling2D
+from keras.layers.normalization import BatchNormalization
+from keras.layers import Input, merge
+from keras.models import Model
+from keras.utils import np_utils
+from keras.callbacks import ModelCheckpoint
+from keras.datasets import cifar10
+from keras.preprocessing.image import ImageDataGenerator
+from keras.regularizers import l2
+from keras.optimizers import RMSprop, SGD
+from sklearn.cross_validation import train_test_split
+import tensorflow as tf
+
+
+# Allocate full of GPU resource to me
+# gpu_options = tf.GPUOptions(per_process_gpu_memory_fraction=1.)
+# sess = tf.Session(config=tf.ConfigProto(gpu_options=gpu_options))
+
+
+def building_residual_block(input_shape, n_feature_maps, kernel_sizes=None, n_skip=2, is_subsample=False, subsample=None):
```

Comment:
This function name sounds weird to me. `build_residual_block` or just `residual_block` would feel more natural.